types/mongodb/index.d.ts

```diff
@@ -103,54 +153,33 @@ export interface SocketOptions {
     socketTimeoutMS?: number;
 }
 
-//http://mongodb.github.io/node-mongodb-native/2.1/api/Server.html
-export interface ServerOptions {
-    // - specify the number of connections in the pool default:5
-    poolSize?: number;
-    ssl?: boolean;
-    sslValidate?: Object;
-    checkServerIdentity?: boolean | Function;
-    sslCA?: Array<Buffer | string>;
-    sslCert?: Buffer | string;
-    sslKey?: Buffer | string;
-    sslPass?: Buffer | string;
-    socketOptions?: SocketOptions;
+// http://mongodb.github.io/node-mongodb-native/2.2/api/Server.html
+export interface ServerOptions extends SSLOptions, HighAvailabilityOptions{
```

Comment:
I've discovered that options from``HighAvailabilityOptions`` interface are legal for all three classes (i.e. [Server][s], [Mongos][m] and [Replset][r]) but a few actually used inside the source code.

[s]: ../../../mongodb/node-mongodb-native/blob/2.2/lib/server.js#L47
[m]: ../../../mongodb/node-mongodb-native/blob/2.2/lib/mongos.js#L50
[r]: ../../../mongodb/node-mongodb-native/blob/2.2/lib/replset.js#L43